### DISCUSSION TITLE
Unattended compilation test with different boards and features using PlatformIO

### DISCUSSION THREAD

**Discussion by aviateur17** [timmbogner/Farm-Data-Relay-System#195] [Ideas]
I've been thinking about how to test code compilation with the different gateway scenarios including different features (ESP-NOW, LoRa, RTC, GPS, OLED, DEBUG) and different micro processor architectures (ESP8266, ESP32, Pico, SAMD21) and I came up with a platformio.ini file that tests different combinations of the above.  It doesn't have every possible combination but many of the common ones.  I'll attach the platformio.ini file that I'm using and the main.cpp file that I use.  If you have any questions let me know and if you want to improve on the platformio.ini file that would be awesome too.  This file only compiles the different gateway possibilities at this time. I haven't added in the controller/sensor combinations yet.

Sample output:
```
Environment                   Status    Duration
----------------------------  --------  ------------
gw_esp32_0_mqtt_gateway       SUCCESS   00:00:44.417
gw_esp32_1_uart_gateway       SUCCESS   00:00:38.168
gw_esp32_2_espnow_repeater    SUCCESS   00:00:46.452
gw_esp32_3_lora_repeater      SUCCESS   00:00:39.714
gw_esp8266_0_mqtt_gateway     SUCCESS   00:00:57.160
gw_esp8266_1_uart_gateway     SUCCESS   00:00:50.437
gw_esp8266_2_espnow_repeater  SUCCESS   00:00:53.309
gw_esp8266_3_lora_repeater    SUCCESS   00:00:51.816
gw_esp32s3_0_mqtt_gateway     SUCCESS   00:00:44.816
gw_esp32s3_1_uart_gateway     SUCCESS   00:00:37.737
gw_esp32s3_2_espnow_repeater  SUCCESS   00:00:39.839
gw_esp32s3_3_lora_repeater    SUCCESS   00:00:38.611
gw_SAMD21_3_lora_repeater     FAILED    00:00:10.785
=============================================================== 1 failed, 12 succeeded in 00:09:13.260 ===============================================================
```
[platformio.ini.txt](https://github.com/timmbogner/Farm-Data-Relay-System/files/14235324/platformio.ini.txt)
[main.cpp.txt](https://github.com/timmbogner/Farm-Data-Relay-System/files/14235326/main.cpp.txt)

**Comment by timmbogner** [comment]
I usually just waste my free time compiling and reconfiguring in Arduino, but this is a much smarter idea😂. I think I've considered a batch file setup with Arduino CLI but never looked into it further.

Seems like I kinda review PIO every year or so, but I'm overdue right now.  I'll get it going and check this out at some point. If you think it's something other people could use, feel free to make a folder in the 'extras' directory for it.
